### DISCUSSION TITLE
Support opening over lockscreen

### DISCUSSION THREAD

**Discussion by AeliusSaionji** [deckerst/aves#1160] [Ideas]
I have aves as my only photos app.

My camera can open over the lockscreen, and I can tap to view a picture I have just taken.

With google photos, this would show the photos taken while the screen was locked.

When google photos is uninstalled, my camera will open aves- but if the phone is locked, then I get presented with the "enter your pin" lockscreen UX instead.

This actually happens a lot, because my phone has a dedicated camera button and I use it all the time to start the camera over the lockscreen.

❤️ aves

**Comment by deckerst** [comment]
It's been discussed in the following issue: 
- https://github.com/deckerst/aves/issues/1052

On my phone (a Sony Xperia), it acts very similarly to your phone (the camera app can only preview with Google Photos and if I disable Photos is uses Aves) but previewing from the lock screen simply fails and shows the lock screen instead. It doesn't ask me to unlock the phone like it does for you.

Anyway, I'd like to support this, but it seems phone manufacturers are so tied to Google Photos that it's difficult to use alternatives, even if it's possible in theory.

**Reply by AeliusSaionji** [reply]
I too have an xperia- I was going from memory and was wrong about it showing the pinpad.

Aves is open under the lockscreen though.

**Comment by AeliusSaionji** [comment]
I don't know if you've seen this, but it might help:

https://github.com/gregordr/Secure-Photo-Viewer

It still works on android 14.

**Reply by deckerst** [reply]
Can you use that app to review pictures you take from the lock screen?

**Reply by AeliusSaionji** [reply]
No, but it may be a code reference for implementing the feature.
You share pictures to it, and the shared pictures appear over the lockscreen next time you lock.

Xperia cameras seem to pass "pictures taken this session" to the default photo app. I suspect many camera apps behave this way. I know some keep a persistent history.

When Aves is invoked, it could first check if the phone is locked. If so, render only the photos aves was invoked with (no album/library access), over the lockscreen- just as the above app does.

This relies on the camera app giving aves the relevant pictures. I don't expect there is a way for aves to find that information otherwise.

**Comment by deckerst** [comment]
I checked the device logs when I disabled Google Photos, launched the Sony camera from the lock screen, took a photo and tried to open the preview. The camera app then sends the [review](com.android.camera.action.REVIEW) intent. I expected something like the [review secure](https://developer.android.com/reference/android/provider/MediaStore#ACTION_REVIEW_SECURE) intent, but anyway Aves can handle both.

Maybe the missing piece is that special activity setting [setShowWhenLocked](https://developer.android.com/reference/android/app/Activity#setShowWhenLocked(boolean)). I need to take a closer look.

**Comment by deckerst** [comment]
Oh, that was it. At least, it works on my device lock screen.
